### PR TITLE
feat(ci): check licenses on PRs

### PR DESCRIPTION
**Pull request by gansheer**
Ref #5631 

* Add a new `licenses` github workflow (existing `validate` workflow has different file list filter)
* Fix the check_licenses.sh script to ensure it result in `exit 1` when an invalid file is detected
* Fix some still missing licenses

**Release Note**
```release-note
feat(ci): check licenses on PRs
```

### CODE REVIEW

File: .github/workflows/licenses.yaml
Comment (squakez) [inline .github/workflows/licenses.yaml:1]: I'd avoid to add another action. We can leverage the lint action and include it there also the license check. In this way, also locally, `make lint` would report any problem.

Comment (squakez) [reply .github/workflows/licenses.yaml:1]: Just seen you mentioned the lint has a different file list. Then, we can probably extend the lint to be more generic.

File: .github/workflows/validate.yml [hunk 1]
```diff
@@ -22,18 +22,24 @@ on:
...
-    paths:
-      - 'pkg/**'
-      - 'cmd/**'
-      - 'addons/**'
+    paths-ignore:
+      - 'proposals/**'
+      - '**.md'
+      - '**.adoc'
+      - 'KEYS'
+      - 'LICENSE'
```
[the remark this answers is not in the document]
Comment (gansheer) [reply .github/workflows/validate.yml:41]: What do you mean exactly ? From what I see we are using these files as headers: https://github.com/apache/camel-k/tree/main/script/headers.

[the remark this answers is not in the document]
Comment (squakez) [reply .github/workflows/validate.yml:41]: Right, sorry, I thought it was taken from `LICENSE`

**Review by squakez — APPROVED** [review]

**Pull request merged**